crates/nu-command/src/system/run_external.rs

```diff
@@ -324,9 +325,34 @@ impl ExternalCommand {
                             }
                         };
 
+                        let mut err_str = err.to_string();
+                        if engine_state.is_interactive {
+                            let mut engine_state = engine_state.clone();
+                            if let Some(hook) = engine_state.config.hooks.command_not_found.clone()
+                            {
+                                if let Ok(PipelineData::Value(Value::String { val, .. }, ..)) =
+                                    eval_hook(
```

Comment:
Would it be running the `command_not_found` hook instead of returning an error if they set it?

I'm imagining something like a distro my add this hook to go find a command and help you install it. Once installed, the hook returns. I think we'd still show an error in that case?

Reply:
It would still return an error. I can see some issues with helping people install packages from errors.

- A single command can be provided by multiple packages.
- Every error about command typos giving a package install interface, feels a bit annoying.

It also doesn't apply to the *Arch Linux* case, as the only way to install things is to manually run *Pacman*.

Reply:
Interesting - can you describe the use case where you want to run a command but still give the error? I can't think of one but maybe you have one in mind

Reply:
Well say I have some pretty long command/script I enter, and within that script there's some typo. In that case I think it's pretty useful to see where the typo is. It's better than the output just telling me where to find, or that this command with a typo in it can't be found.

Reply:
Oh okay, got it. Yeah now that I see how it works together that makes sense.